### DISCUSSION TITLE
Single window only

### DISCUSSION THREAD

**Discussion by midiwidi** [Xpra-org/xpra-html5#242] [Q&A] [answered]
I have a local application which talks to a piece of hardware through a serial port.
I would like colleagues to control that piece of hardware remotely using Xpra and the html5 extension.
In general, the native Xpra client / server connection as well as the html5 access through the browser from a remote machine work,

On my local machine I start the server with `xpra start --bind-tcp=0.0.0.0:10000 --sharing=yes --html=on --start-child='pipenv run python OBC_Emulator.py'`
If I run the native client I on a remote machine I get a singe window (just as I want to), but if I run it through a browser, then I get this menu from which I can start every program I want. I don't want my colleagues to start a random application of my machine. The menu can be turned off, but this setting is made on the client side. I don't see a way to configure that from the server.

Is it possible to run my application exclusively and prevent my colleagues from starting other applications while using the html5 client?

**Comment by totaam** [comment, marked as answer]
`--start-new-commands=no`

**Reply by midiwidi** [reply]
Thank you for that. Is there also an option to prevent the client to shutdown the server?
I didn't find anything in the man page.

**Reply by totaam** [reply]
Try: `xpra start --env=XPRA_CLIENT_CAN_SHUTDOWN=0 ...`

**Reply by midiwidi** [reply]
That kind of works. The client menu still shows the shutdown option, but the shutdown is ignored which is what I wanted.
Unfortunately `xpra stop` on the server is also ignored. To work around that I can whether kill xpra or run it with --daemon=no and use CTRL+C.

**Reply by totaam** [reply]
> The client menu still shows the shutdown option

Fixed: b30082d2dee31ba4ae21e14762ff72cb9c565b96

> Unfortunately `xpra stop` on the server is also ignored

Yes, you can't have it both ways. `xpra stop` is a client.

> To work around that I can whether kill xpra or run it with --daemon=no and use CTRL+C

This is dangerous as the server will be shutdown if anything happens to the controlling session - it also causes problems with the vfb occasionally receiving the signals intended for the xpra server which prevents a clean shutdown.

A much safer option is to send a SIGINT to the server process.
You can find the server pid with:
```shell
xpra info | grep server.pid=
```
(with newer versions of xpra)

Or for older versions:
```shell
xpra info | grep server.pidfile.path=
```
(the pid is in that file)
